src/js/component.js

```diff
@@ -134,9 +137,6 @@ class Component {
     this.childIndex_ = null;
     this.childNameIndex_ = null;
 
-    // Remove all event listeners.
-    this.off();
```

Comment:
Yes, it does. It's part of why the mixin is applied to instances instead of prototypes (to answer your other question).

It's a toss-up. You can have this `off()` on `dispose` behavior built-in or you can apply the mixin to a prototype and manually call `this.off()` in the `dispose` method of that class.